odb_extractor_atoms.py

```diff
@@ -6,15 +6,179 @@
+           'GMOS-S': 'instrument:fpu', 'NIRI': 'instrument:mask'}
+
+gpi_filter_wav = {'Y': 1.05, 'J': 1.25, 'H': 1.65, 'K1': 2.05, 'K2': 2.25}
+nifs_filter_wav = {'ZJ': 1.05, 'JH': 1.25, 'HK': 2.20}
+
+
+def find_filter(input, filter_dict):
+    """Match input string with filter list (in dictionary)"""
+
+    filter = ''
+    filters = list(filter_dict.keys())
+    for filt in filters:
+        if filt in input:
+            filter = filt
+            break
+    return filter
+
+
+def uniquelist(seq):
+    """Make a list of unique values"""
+    # http://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-in-python-whilst-preserving-order
+    seen = set()
+    seen_add = seen.add
+    return [x for x in seq if not (x in seen or seen_add(x))]
+
+
+def searchlist(val, alist):
+    """Search for existence of val in any element of alist"""
+    found = False
+    for elem in alist:
+        if val in elem:
+            found = True
+            break
+    return found
+
+
+def shortid(idin):
+    """Return short form of obsid or data label"""
+    idsp = idin.split('-')
+    #         print(obsidsp)
+    idout = idsp[0][1] + idsp[1][2:5] + '-' + idsp[2] + '-' + idsp[3] + '[' + idsp[4] + ']'
+    if len(idsp) == 6:
+        idout += '-' + idsp[5]
+    return idout
+
+
+def odb_json(progid, path='None', overwrite=False, verbose=False):
+    """
+    Download json of ODB program information
+
+    Parameters
+        progid:  Program ID of program to extract
+        path:    Path for json files
+        overwrite: Overwrite any existing json files?
+        verbose: Verbose output?
+
+    Return
+        json_result:   JSON query result as a list of dictionaries
+    """
+
+    if progid == "":
+        print('odb_json: program id not given.')
+        raise ValueError('Program id not given.')
+
+    file = progid + '.json.gz'
+    if not overwrite and path != 'None' and os.path.exists(os.path.join(path, file)):
+        with gzip.open(os.path.join(path, file), 'r') as fin:
+            json_bytes = fin.read()
+
+        json_str = json_bytes.decode('utf-8')
+        json_result = json.loads(json_str)
+    else:
+        response = requests.get(
+            'http://gnodbscheduler.hi.gemini.edu:8442/programexport?id=' + progid)
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as exc:
+            print('odb_json: request failed: {}'.format(response.text))
+            raise exc
+        else:
+            json_result = response.json()
+            if (overwrite or not os.path.exists(os.path.join(path, file))) and path != 'None':
+                json_str = json.dumps(json_result, indent=2)
+                with gzip.open(os.path.join(path, file), 'wb') as fout:
+                    fout.write(json_str.encode('utf-8'))
+
+        if verbose:
+            print(response.url)
+        # print(response.text)
+
+    return json_result
+
+# --------------
+
+
+def obsmode(config):
```

Comment:
I would think we would want `mode` to be an `Enum` and not a `str`?

```
class ObservationMode(str, Enum):
    UNKNOWN = 'unknown'
    IMAGING = 'imaging'
    LONGSLIT = 'longslit'
    IFU = 'ifu'
    MOS = 'mos'
    XD = 'xd'
    CORON = 'coron'
    NRM = 'nrm'

    @staticmethod
    def obsmode(config):
        # Put all obsmode logic in here
        ...
```

Then we have all of this self-contained, and can just call:
```
ObservationMode.obsmode(config)
```
which would be nice and clean.

Reply:
Also, do we want `unknown` or could we use `None` instead?